app/services/service_offering.rb

```diff
@@ -15,5 +15,8 @@ def show(id)
     @service_offering_ref = id
     obj = api_instance.show_service_offering(id)
     apply_instance_vars(obj)
+  rescue TopologicalInventoryApiClient::ApiError => e
+    Rails.logger.error("TopologicalInventoryApiClient::ApiError #{e.message}")
+    raise ActiveRecord::RecordNotFound
```

Comment:
@syncrou 
What should we do here if we can't talk to Topology Service, if its down or if the service offering is invalid.
For now I am throwing a RecordNotFound should be throwing  a 5xx error

Reply:
I think a '503' `:service_unavailable` would make the most sense.

Reply:
@bdunne - Does topology raise a 503 when it is down? Then we could rescue on that and then re raise a 503 for those specific instances and a 404 or what we deem as correct for other TopologyApi errors.

Reply:
@syncrou If the Server is not there (bad url) or there is an exception on the topology service we get a single exception, its up to us on how we translate it.

Reply:
Yes, but if OCP knows that the Topology api is down, there is a pattern for returning a static page which we could read as a 503.  I'm just suggesting that if we know the service is down we should raise a 503 to our users as well, which would help us troubleshoot.

Reply:
@mkanoor What happens if we don't rescue this error  and let it raise to the caller?  Or log and re-raise the error?

Reply:
@mkanoor It raises the HTTP 500 error indicating there was a service side error

Reply:
I like that idea. We can follow that up in another PR.